File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/template/EmailVerificationTemplateCreator.java
package com.eager.questioncloud.core.domain.verification.template;

import com.eager.questioncloud.core.domain.verification.enums.EmailVerificationType;
import com.eager.questioncloud.core.exception.CoreException;
import com.eager.questioncloud.core.exception.Error;

public class EmailVerificationTemplateCreator {
    public static EmailVerificationTemplate getTemplate(EmailVerificationType emailVerificationType, String token) {
        switch (emailVerificationType) {
            case CreateUser -> {
                return new CreateUserEmailVerificationTemplate(token);
            }
            case ChangePassword -> {
                return new ChangePasswordEmailVerificationTemplate(token);
            }
            default -> {
                throw new CoreException(Error.INTERNAL_SERVER_ERROR);
            }
        }
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/verification/template/EmailVerificationTemplateCreator.kt
package com.eager.questioncloud.core.domain.verification.template

import com.eager.questioncloud.core.domain.verification.enums.EmailVerificationType
import com.eager.questioncloud.core.exception.CoreException
import com.eager.questioncloud.core.exception.Error

object EmailVerificationTemplateCreator {
    fun getTemplate(emailVerificationType: EmailVerificationType, token: String): EmailVerificationTemplate {
        return when (emailVerificationType) {
            EmailVerificationType.CreateUser -> {
                CreateUserEmailVerificationTemplate(token)
            }

            EmailVerificationType.ChangePassword -> {
                ChangePasswordEmailVerificationTemplate(token)
            }

            else -> {
                throw CoreException(Error.INTERNAL_SERVER_ERROR)
            }
        }
    }
}
